src/iosched/unified.c

```diff
@@ -880,8 +947,10 @@ ssize_t unified_write(struct dentry *d, const char *buf, size_t size, off_t offs
 		}
 	}
 	ltfs_mutex_unlock(&d->iosched_lock);
-	if (spare_cache)
+	if (spare_cache) {
+		_unified_put_dentry_priv(dpr, priv);
```

Comment:
Isn't there a missing test for `if (dpr)` before the call to `_unified_put_dentry_priv`?

Reply:
This `_unified_put_dentry_priv` call is not needed because L931-L933 didi that already. Fixed in my branch.